### DISCUSSION TITLE
v2.3 - Unable to run so-wazuh-agent-manage

### DISCUSSION THREAD

**Discussion by bahoogawooga** [Security-Onion-Solutions/securityonion#6322] [Unsupported Versions] [answered]
fresh install of v2.3, trying to add wazuh agents and encountering the following

```
sudo so-wazuh-agent-manage
2021/11/23 02:08:37 manage_agents: CRITICAL: (1226): Error reading XML file '/var/ossec/etc/ossec.conf':  (line 0).
```

log file (/nsm/wazuh/logs/osssec.log) show the following

     66 2021/11/22 23:38:36 ossec-logcollector: INFO: (1904): File not available, ignoring it: '/var/ossec/logs/active-responses.log'.
     67 2021/11/22 23:48:34 ossec-csyslogd: CRITICAL: (1226): Error reading XML file '/var/ossec/etc/ossec.conf':  (line 0).
     68 2021/11/22 23:48:34 ossec-authd: CRITICAL: (1226): Error reading XML file '/var/ossec/etc/ossec.conf':  (line 0).
     69 2021/11/22 23:53:12 ossec-csyslogd: CRITICAL: (1226): Error reading XML file '/var/ossec/etc/ossec.conf':  (line 0).
     70 2021/11/22 23:53:12 ossec-authd: CRITICAL: (1226): Error reading XML file '/var/ossec/etc/ossec.conf':  (line 0).
     71 2021/11/23 01:54:53 manage_agents: CRITICAL: (1226): Error reading XML file '/var/ossec/etc/ossec.conf':  (line 0).
     72 2021/11/23 02:03:02 manage_agents: CRITICAL: (1226): Error reading XML file '/var/ossec/etc/ossec.conf':  (line 0).
     73 2021/11/23 02:08:32 manage_agents: CRITICAL: (1226): Error reading XML file '/var/ossec/etc/ossec.conf':  (line 0).

seems to be two ossec.conf file.  one is located

`'/var/ossec/etc/ossec.conf'`

and the other is at

`/opt/so/conf/wazuh/ossec.conf`

running xmllint shows the following

```
 sudo xmllint /var/ossec/etc/ossec.conf
/var/ossec/etc/ossec.conf:167: parser error : Extra content at the end of the document
<ossec_config>
```

if you look at the /var/ossec/etc/ossec.conf file, it seems to have an extra <ossec_config> and </ossec_config> tag at the end of the file

    152
    153   
    154   <active-response>
    155     <disabled>no</disabled>
    156     <ca_store>/var/ossec/etc/wpk_root.pem</ca_store>
    157     <ca_verification>yes</ca_verification>
    158   </active-response>
    159
    160   
    161   <logging>
    162     <log_format>plain</log_format>
    163   </logging>
    164
    **165 </ossec_config>
    166
    167 <ossec_config>
    168   <localfile>
    169     <log_format>syslog</log_format>
    170     <location>/var/ossec/logs/active-responses.log</location>
    171   </localfile>
    172   <localfile>
    173     <log_format>syslog</log_format>
    174     <location>/var/log/secure</location>
    175   </localfile>
    176   <localfile>
    177     <log_format>syslog</log_format>
    178     <location>/var/log/syslog</location>
    179   </localfile>
    180
    181   <localfile>
    182     <log_format>syslog</log_format>
    183     <location>/var/log/dpkg.log</location>
    184   </localfile>
    185
    186   <localfile>
    187     <log_format>syslog</log_format>
    188     <location>/var/log/kern.log</location>
    189   </localfile>
    190
    191 </ossec_config>**

**Comment by dougburks** [comment, marked as answer]
Thanks for reporting this @bahoogawooga!

I've created the following issue:
https://github.com/Security-Onion-Solutions/securityonion/issues/6325

In the meantime, you should be able to manually remove the xml header on the first line of `/opt/so/conf/wazuh/ossec.conf` to get `so-wazuh-agent-manage` running again.

**Reply by bahoogawooga** [reply]
thanks @dougburks,

i don't think the error message 

`CRITICAL:` (1226): Error reading XML file '/var/ossec/etc/ossec.conf': (line 0).

 is accurate.  the issue seems to begin at line 167 where there is an extra tag

`<ossec_config>`

I'll try removing the extra tags and report back.

here's a thread i found on another github where someone ran into something similar regarding the error message not being very accurate on which line the error lives fwiw.  

- ![image](https://user-images.githubusercontent.com/94880164/143060157-c3e161c4-c600-4f62-8851-57d409d55d73.png)

**Reply by dougburks** [reply]
This issue is complicated by Docker mapping. When so-wazuh-agent-manage says `Error reading XML file '/var/ossec/etc/ossec.conf`, it's referring to the file inside the Docker container which is actually mapped to /opt/so/conf/wazuh/ossec.conf on the host filesystem. A bug in 2.3.90 is causing that file to incorrectly have an XML header. Our testing has confirmed that removing the XML header allows Wazuh to work correctly again. We are working on finalizing the fixes now.

**Reply by dougburks** [reply]
Security Onion 2.3.90 WAZUH Hotfix Now Available!
https://blog.securityonion.net/2021/11/security-onion-2390-wazuh-hotfix-now.html
